logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -193,7 +203,9 @@ private void innerWriteEntry(DLQEntry entry) throws IOException {
         int eventPayloadSize = RECORD_HEADER_SIZE + record.length;
         if (currentQueueSize.longValue() + eventPayloadSize > maxQueueSize) {
             if (storageType == QueueStorageType.DROP_NEWER) {
-                logger.error("cannot write event to DLQ(path: " + this.queuePath + "): reached maxQueueSize of " + maxQueueSize);
+                logger.error(lastError);
+                lastError = String.format("Cannot write event to DLQ(path: %s): reached maxQueueSize of %d", queuePath, maxQueueSize);
```

Comment:
These need to be switched, I think.
```suggestion
                lastError = String.format("Cannot write event to DLQ(path: %s): reached maxQueueSize of %d", queuePath, maxQueueSize);
                logger.error(lastError);
```